internal/envoy/v3/cluster.go

```diff
@@ -187,8 +193,13 @@ func DNSNameCluster(c *dag.DNSNameCluster) *envoy_cluster_v3.Cluster {
 
 	cluster.Name = envoy.DNSNameClusterName(c)
 	cluster.DnsLookupFamily = parseDNSLookupFamily(c.DNSLookupFamily)
+
+	clusterType := envoy_cluster_v3.Cluster_STRICT_DNS
+	if cluster.DnsLookupFamily == envoy_cluster_v3.Cluster_ALL {
+		clusterType = envoy_cluster_v3.Cluster_LOGICAL_DNS
+	}
 	cluster.ClusterDiscoveryType = &envoy_cluster_v3.Cluster_Type{
```

Comment:
nit: could use `ClusterDiscoveryType` helper here rather than constructing it ourselves

I kinda like the setup you have here vs above so we don't have to construct a `*envoy_cluster_v3.Cluster_Type` multiple times and rather just swap out the type value itself